CounterStrike/src/dev/danablend/counterstrike/utils/MyBukkit.java

```diff
@@ -0,0 +1,111 @@
+package dev.danablend.counterstrike.utils;
+
+import dev.danablend.counterstrike.CounterStrike;
+import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.scheduler.BukkitTask;
+
+import static org.bukkit.Bukkit.getServer;
+import static org.bukkit.Bukkit.isOwnedByCurrentRegion;
+
+public class MyBukkit {
+    private CounterStrike main;
+    private boolean folia ;
+
+
+    public MyBukkit(CounterStrike main) {
+        this.main = main;
+
+        String version = Bukkit.getVersion().toUpperCase();
+        folia = (version.contains("-FOLIA-"));
+
+        //server type name
+        if (!folia) {
+            String minecraftVersion2 = Bukkit.getServer().getName();
+            folia = (minecraftVersion2.toUpperCase().contains("FOLIA"));
+        }
+    }
+
+
+    public boolean isFolia() {
+        return folia;
+    }
+
+
+    public Object runTask(Player player, Location local, Entity entity, Runnable myrun) {
```

Comment:
I think all operations on this Object are done in this class, so instead of returning object, you could return a custom object, that would be sth like

```
class CSMCTask {
  BukkitTask taskBukkit = null;
  ScheduledTask scheduledTask = null;
}
```

and depending on the bukkit engine used, access one field or another. You would not need to cast Object to tasks, return Object to another classes, which does not look great. Returning typed object looks better

Reply:
On the second thought, having in this class ScheduledTask might not be possible, as it might throw an error on spigot, which will not know, what is this type. In that case it might be 
```
class CSMCTask {
  BukkitTask taskBukkit = null;
  Object scheduledTask = null;
}
```
This is not that great, but in other classes you would use type CSMCTask instead of Object

Reply:
Been thinking from time to time about it, for now, considering my limitations, this seems the best option

Later on it could implement something like a dictionary and manage internally these objects from a general key